.github/workflows/ci.yaml

```
@@ -2,24 +2,42 @@ name: ci
 
 on: [push, pull_request]
 
+env:
+  PYTHON_VERSION: "3.11"
+  GO_VERSION: "1.22"
+
 jobs:
   validate:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: [3.x]
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
+        uses: actions/checkout@v4
       - name: Setup Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v5
         with:
-          python-version: 3.x
+          python-version: ${{ env.PYTHON_VERSION }}
       - name: Download dependencies
         run: |
           pip3 install -r requirements.txt
       - name: Linters
         run: |
           pre-commit run --all-files
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Download dependencies
+        run: |
+          pip3 install -r requirements.txt
+      - name: Test
+        run: |
+          echo "To be done"
```

I think it would be safe to say `pytest` here. That's what I would like at least - to have all tests executed just by running `pytest`. We can add bells and whistles (e.g. coverage report) later.

```suggestion
    pytest
```

---

Job will fail as we do not have test right now. I have checked and exit code is 5. I did not want to merge things with a CI failing.

---

Ack. Sorry.

---

P.S. Kind of confusing that `pytest` output looks happy but the exit status is non-zero.

---

Yes, but if you run `pytest` I think it is expected to find some tests. If you do not find any test file at all, I guess it is probably an issue with the user invocation and thus the error code. But just my guess.